src/compiler/c.rs

```diff
@@ -314,19 +291,24 @@ impl<T, I> CompilerHasher<T> for CCompilerHasher<I>
 }
 
 impl<I: CCompilerImpl> Compilation for CCompilation<I> {
-    fn generate_compile_commands(&self)
+    fn generate_compile_commands(&self, path_transformer: &mut dist::PathTransformer)
                                 -> Result<(CompileCommand, Option<dist::CompileCommand>, Cacheable)>
     {
         let CCompilation { ref parsed_args, ref executable, ref compiler, preprocessed_input: _, ref cwd, ref env_vars } = *self;
-        compiler.generate_compile_commands(executable, parsed_args, cwd, env_vars)
+        compiler.generate_compile_commands(path_transformer, executable, parsed_args, cwd, env_vars)
     }
 
-    fn into_inputs_creator(self: Box<Self>) -> Result<Box<FnMut(&mut io::Write)>> {
+    fn into_dist_inputs_creator(self: Box<Self>, path_transformer: &mut dist::PathTransformer) -> Result<Box<FnMut(&mut io::Write)>> {
+        let input_path = self.cwd.join(&self.parsed_args.input);
+        let dist_input_path = path_transformer.to_dist(&input_path).unwrap();
+        // tar-rs imposes that `set_path` takes a relative path
+        assert!(dist_input_path.starts_with("/"));
```

Comment:
These shouldn't be actual `assert!`s, should they? Those will take down the whole server.